File: 3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.cpp
class Solution {
public:
    string smallestPalindrome(string s) {
        int n=s.size();
    if(s.size()%2!=0){
        sort(s.begin(),s.begin()+n/2);
        sort(s.begin()+n/2+1,s.end(), greater<int>());

    } 
    else{
      sort(s.begin(),s.begin()+n/2);
        sort(s.begin()+n/2,s.end(), greater<int>());   
    }
    return s;
    }
};

File: 3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.java
class Solution {
    public String smallestPalindrome(String s) {
    int n = s.length();
    String first = s.substring(0,n/2);
    char[] sort = first.toCharArray();
    Arrays.sort(sort);
    String left = new String(sort);
    //if n is odd
    String middle = (n % 2 == 1) ? "" + s.charAt(n / 2) : "";
    //reverse left as it is pallindrome and convert to string
    String right = new StringBuilder(left).reverse().toString();
    return left+middle+right;
    }
}